CRM/Financial/BAO/PaymentProcessor.php

```diff
@@ -184,20 +131,79 @@ public static function del($paymentProcessorID) {
     static::deleteRecord(['id' => $paymentProcessorID]);
   }
 
+  /**
+   * Callback for hook_civicrm_pre().
+   * @param \Civi\Core\Event\PreEvent $event
+   */
+  public static function self_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
+    if ($event->action === 'create') {
+      // If we are creating a new PaymentProcessor and have not specified the payment instrument to use, get the default from the Payment Processor Type.
+      if (empty($event->params['payment_instrument_id'])) {
+        $event->params['payment_instrument_id'] = civicrm_api3('PaymentProcessorType', 'getvalue', [
+          'id' => $event->params['payment_processor_type_id'],
+          'return' => 'payment_instrument_id',
+        ]);
+      }
+      // Supply defaults for `title` and `frontend_title`
+      if (!isset($event->params['title'])) {
+        $event->params['title'] = $event->params['name'];
+      }
+      if (!isset($event->params['frontend_title'])) {
```

Comment:
Ah ok - this was the bit I thought was going into `writeRecord` - but yeah it doesn't have to